### ISSUE TITLE
format of record keys

### ISSUE THREAD

**Issue by r-tent** [sdcTools/UserSupport#105]
*Please specify*  
**SDC tool used:** tau-argus, sdcTable
**Version used:**  4.1.12BETA
**Operating system used:** Windows

---

Hello @ppdewolf,

sorry for getting on your nerves again, but I have one further remark of something that took me a while, when testing the cell key method of tau-Argus. Even though it was more a problem of myself than of your program, maybe you could fix the following "issue".

When I light heartedly added record keys to my data via R , I didn't notice that some values where stored in a form like 6e-05 instead of 0.00006. Now for tau-Argus, this was quite a big deal while for me it was rather untraceable, since some csv-viewiers convert this expression directly, when opening the file.
I think it would be great if tau-Argus accepted values like 6e-05 for record keys :-)

**Comment by ppdewolf** [comment]
Good point. Something to add in a newer version or at least something to mention in the quick reference and manual.

**Comment by ppdewolf** [comment]
Are you using fixed format or free format input?

Do you have the same problem when using a response variable in that format (i.e., e.g., turnover 6.7861e4)?

**Comment by ppdewolf** [comment]
This seems to occur only with **free format**. Happens with any numeric variable (both very small and very large numbers).

**Work around for record keys**
I've updated the example how to produce the record keys in R such that the scientific notation is not used and the resulting numbers all have 15 decimals. Something like this:
`# Record keys (direct approach)`
`microdata <- read.csv2(file=”hc_9_2_synth.csv”)`
`set.seed(123)`
`microData$recordkey <- format(runif(dim(microData)[1]), scientific=FALSE, nsmall=15)`
`write.table(microData, ”hc_9_2_synth_Tau.csv”, row.names=FALSE, col.names=FALSE, quote=FALSE, dec=”.”,sep=”;”)`
